pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -342,8 +348,12 @@ public synchronized boolean canUnsubscribe(Consumer consumer) {
     }
 
     @Override
-    public CompletableFuture<Void> close() {
+    public synchronized CompletableFuture<Void> close() {
```

Comment:
do we need synchronized over the whole thing? surely it's just needed over the new code.

Reply:
Yeah, we don't need it on the whole method. Just making sure the instance of tracker is correctly updated. I'll restrict the sync block

Reply:
@merlimat this is still over synchronizing

Reply:
Fixed this one